http/src/main/java/org/sapia/ubik/rmi/server/transport/http/HttpTransportProvider.java

```diff
@@ -82,14 +85,28 @@ public synchronized Connections getPoolFor(ServerAddress address) throws RemoteE
 
     if ((conns = pools.get(address)) == null) {
       try {
+        int maxConnections = Conf.getSystemProperties().getIntProperty(HTTP_CLIENT_MAX_CONNECTIONS_KEY, DEFAULT_MAX_CLIENT_CONNECTIONS);
         if (usesJakarta) {
-          int maxConnections = Conf.getSystemProperties().getIntProperty(HTTP_CLIENT_MAX_CONNECTIONS_KEY, DEFAULT_MAX_CLIENT_CONNECTIONS);
           conns = new HttpClientConnectionPool((HttpAddress) address, maxConnections);
         } else {
-          conns = new JdkClientConnectionPool((HttpAddress) address);
+          conns = new JdkClientConnectionPool((HttpAddress) address, maxConnections);
         }
 
         pools.put(address, conns);
+        
+        Hub.getModules().getTaskManager().addTask(new TaskContext("TimedOutHttpConnectionReaper", 
+            HttpConsts.DEFAULT_TIMED_OUT_CONNECTION_CLEANING_INTERVAL), new Task() {
+          @Override
+          public void exec(TaskContext ctx) {
+            pools.values().forEach(pool -> {
+              if (pool instanceof JdkClientConnectionPool) {
```

Comment:
no special logic for pool of type HttpClientConnectionPool? ... don't have (yet) the full context of your change, so my question might be stupid

Reply:
no, I was thinking about generalizing this but for now only HttpUrlConnection is causing us problems.

Reply:
Then I assume that HttpClientConnectionPool is not using HttpUrlConnection?